base/applications/mspaint/main.cpp

```diff
@@ -115,193 +82,171 @@ OFNHookProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
     return 0;
 }
 
-/* entry point */
+BOOL CMainWindow::DoGetOpenFileName(IN OUT LPTSTR pszFile, INT cchMaxFile)
+{
+    static OPENFILENAME ofn = { 0 };
+    static CString strFilter;
+
+    if (ofn.lStructSize == 0)
+    {
+        // The "All Files" item text
+        CString strAllPictureFiles;
+        strAllPictureFiles.LoadString(hProgInstance, IDS_ALLPICTUREFILES);
+
+        // Get the import filter
+        CSimpleArray<GUID> aguidFileTypesI;
+        CImage::GetImporterFilterString(strFilter, aguidFileTypesI, strAllPictureFiles,
+                                        CImage::excludeDefaultLoad, _T('\0'));
+
+        // Initializing the OPENFILENAME structure for GetOpenFileName
+        ZeroMemory(&ofn, sizeof(ofn));
+        ofn.lStructSize    = sizeof(ofn);
+        ofn.hwndOwner      = m_hWnd;
+        ofn.hInstance      = hProgInstance;
+        ofn.lpstrFilter    = strFilter;
+        ofn.Flags          = OFN_EXPLORER | OFN_HIDEREADONLY;
+        ofn.lpstrDefExt    = L"png";
+    }
+
+    ofn.lpstrFile = pszFile;
+    ofn.nMaxFile  = cchMaxFile;
+    return ::GetOpenFileName(&ofn);
+}
+
+BOOL CMainWindow::DoGetSaveFileName(IN OUT LPTSTR pszFile, INT cchMaxFile)
+{
+    static OPENFILENAME sfn = { 0 };
+    static CString strFilter;
 
-int WINAPI
-_tWinMain (HINSTANCE hThisInstance, HINSTANCE hPrevInstance, LPTSTR lpszArgument, INT nCmdShow)
+    if (sfn.lStructSize == 0)
+    {
+        // Get the export filter
+        CSimpleArray<GUID> aguidFileTypesE;
+        CImage::GetExporterFilterString(strFilter, aguidFileTypesE, NULL,
+                                        CImage::excludeDefaultSave, _T('\0'));
+
+        // Initializing the OPENFILENAME structure for GetSaveFileName
+        ZeroMemory(&sfn, sizeof(sfn));
+        sfn.lStructSize    = sizeof(sfn);
```

Comment:
please remove the extra spaces at the left of the equals signs.

Reply:
Removed.